test_v01/test_null_array.cpp

```diff
@@ -0,0 +1,99 @@
+// Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <numeric>
+
+#include "sparrow_v01/layout/null_array.hpp"
+
+#include "doctest/doctest.h"
+
+#include "../test/external_array_data_creation.hpp"
+
+namespace sparrow
+{
+    arrow_proxy make_proxy_for_null_array(std::size_t size)
+    {
+        ArrowSchema sc;
+        ArrowArray ar;
+        test::fill_schema_and_array<null_type>(sc, ar, size, 0u, {});
+        return arrow_proxy(std::move(ar), std::move(sc));
```

Comment:
Static analysis points to this call as not really a move (indeed it copies) but we have different constructors overloads that will not be the same if remove these calls. The warning `performance-move-const-arg` is maybe useful for some case of performance checks but I fear it's pointing at the wrong problem here, it cant check where the moved-form object is passed and if that would change the called overload if we removed these explicit moves. I suspect it will trigger a lot when using `arrow_proxy`. Should we deactivate that check?

Reply:
Actually it should move and not copy the structures. I'll fix the implementation of arrow_proxy and we can keep this warning for now.